Link the signal health-depleted to game scene

ligamesh123 opened the pull request:
I add the signal health_depleted that when player out of HP so the game_scene will change to game over scene.

ligamesh123 committed:
Link the signal health-depleted to game scene

I add the signal health_depleted that when player out of HP the game_scene will change to game over scene

ligamesh123 committed:
Merge branch 'xixs-students:main' into Add-function-to-activate-GameOver-scene-when-player-died

ligamesh123 committed:
Add script game.gd to game.tscn

lawin666 commented:
it's not completed yet because it doesn't change the scene from game scene to game over scene when the player's hp is 0.

ligamesh123 committed:
Merge branch 'xixs-students:main' into Add-function-to-activate-GameOver-scene-when-player-died

ligamesh123 committed:
Connect the signal to game.gd

Connect the signal when player's HP is = 0 so change the scene to GameOver scene

ligamesh123 commented:
I have fixed the code , now the fuction is working.

ExclusiveKnight commented:
The game over scene is not meant to be switched to because of it's transparent background (well it will work fine but the restart button will reset the current scene).
It's better if you add the game over scene as a child to the game scene, and from there you can select the "Player" node in the game scene and select **Node** then click on **Signals**. Signals are better in this case because they can help organize code and you can specifically choose which node (with a script attached) will do if it gets that signal in it's own code.

1. First you need to instantiate GameOver.tscn to game.tscn
![image](https://github.com/xixs-students/IT23-game/assets/57048823/15454184-de36-4a65-b79d-34403d9038f3)

![image](https://github.com/xixs-students/IT23-game/assets/57048823/aa33eceb-0808-4f4d-a32b-e18f2bca1de1)


When you are about to connect the **health_depleted()** signal, connect it to the **Game Over** scene.

![image](https://github.com/xixs-students/IT23-game/assets/57048823/346f8c20-3d41-405f-9e15-1fc8a4e3a658)

Once it's connected, write this code as shown below (I tested it, don't worry):
![image](https://github.com/xixs-students/IT23-game/assets/57048823/37514017-e33f-4215-837a-8aa9d04abeb6)
